OpenSim/Simulation/Test/testWrappingMath.cpp

```diff
@@ -0,0 +1,139 @@
+/* -------------------------------------------------------------------------- *
+ *                         OpenSim:  testWrappingMath.cpp                     *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2020 Stanford University and the Authors                *
+ * Author(s): Nicholas Bianco                                                 *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#define CATCH_CONFIG_MAIN
+#include "Vendors/tropter/external/catch/catch.hpp"
+#include <OpenSim/Simulation/Wrap/WrapMath.h>
+#include <OpenSim/Common/Mtx.h>
+
+#include <random>
+
+using namespace OpenSim;
+
+static std::minstd_rand rng{std::random_device{}()};
+static std::uniform_real_distribution<double> dist;
+static auto generate_v = []{ return dist(rng); };
+
+template<typename C, typename D, int N = 3>
+static void fillMat(C mat1, D mat2) {
+    for (int i = 0; i < N; ++i) {
+        for (int j = 0; j < N; ++j) {
+            double v = generate_v();
+            mat1[i][j] = v;
+            mat2[i][j] = v;
+        }
+    }
+}
+
+template<typename C, typename D, int N = 3>
+static void fillVec(C vec1, D vec2) {
+    for (int i = 0; i < N; ++i) {
+        double v = generate_v();
+        vec1[i] = v;
+        vec2[i] = v;
+    }
+}
+
+template<typename C, typename D, int N = 3>
+static void compareMat(C mat1, D mat2) {
+    for (int i = 0; i < N; ++i) {
+        for (int j = 0; j < N; ++j) {
+            CHECK(mat1[i][j] == Approx(mat2[i][j]).margin(1e-6));
```

Comment:
Why was the margin for this comparison dropped three orders of magnitude to `10^-6`? `double`-precision numbers can be accurate to (e.g.) `10^-16`, but can incur error when multiplying/adding (esp. if the magnitude of those numbers differs by a lot). 

Is there something fundamentally different about this, or the previous, matrix multiplication that causes error to increase by *a lot*? I can understand having somewhat liberal error margins on (e.g.) a converging algorithm, moco, etc. but it's inherently suspicious that we can only multiply `double`-based matrices with 1ppm accuracy (either before or after).

Is this because one approach uses `float`-based trigonometric functions for the rotation?

Reply:
I'm reverting back to the `SimTK_TEST` macros here. These macros use the `SimTK::Test::numericallyEqual()` comparison functions underneath which scales the equality test by the magnitude of the values being compared as to not ask for unreasonable precision. 

We use should probably use these SimTK testing macros over the Catch stuff for any meaningful numerical comparisons (I got a little carried away creating these tests using Catch).